index_documents_test.go

```diff
@@ -1525,30 +1523,95 @@ func TestIndex_GetDocuments(t *testing.T) {
 			},
 		},
 		{
-			name: "TestIndexGetDocumentsWithLimit",
+			name: "TestIndexGetDocumentsWithFields",
 			args: args{
-				UID:    "TestIndexGetDocumentsWithNoExistingDocument",
 				client: defaultClient,
 				request: &DocumentsQuery{
 					Fields: []string{"title"},
 				},
 				resp: &DocumentsResult{},
 			},
 		},
+		{
+			name: "TestIndexGetDocumentsWithFilterAsString",
+			args: args{
+				client: defaultClient,
+				request: &DocumentsQuery{
+					Filter: "book_id = 123",
+				},
+				resp: &DocumentsResult{},
+				filter: []string{
+					"book_id",
+				},
+			},
+			result: 1,
+		},
+		{
+			name: "TestIndexGetDocumentsWithFilterAsArray",
+			args: args{
+				client: defaultClient,
+				request: &DocumentsQuery{
+					Filter: []string{"tag = Tragedy"},
+				},
+				resp: &DocumentsResult{},
+				filter: []string{
+					"tag",
+				},
+			},
+			result: 3,
+		},
+		{
+			name: "TestIndexGetDocumentsWithMultipleFilterWithArrayOfString",
+			args: args{
+				client: defaultClient,
+				request: &DocumentsQuery{
+					Filter: []string{"tag = Tragedy", "book_id = 742"},
+				},
+				resp: &DocumentsResult{},
+				filter: []string{
+					"tag",
+					"book_id",
+				},
+			},
+			result: 1,
+		},
+		{
+			name: "TestIndexGetDocumentsWithMultipleFilterWithInterface",
+			args: args{
+				client: defaultClient,
+				request: &DocumentsQuery{
+					Filter: []interface{}{[]string{"tag = Tragedy", "book_id = 123"}},
+				},
+				resp: &DocumentsResult{},
+				filter: []string{
+					"tag",
+					"book_id",
+				},
+			},
+			result: 4,
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			c := tt.args.client
-			i := c.Index(tt.args.UID)
+			i := c.Index("indexUID")
 			t.Cleanup(cleanup(c))
-			SetUpBasicIndex(tt.args.UID)
+			SetUpIndexForFaceting()
+
+			if tt.args.request != nil && tt.args.request.Filter != nil {
+				gotTask, err := i.UpdateFilterableAttributes(&tt.args.filter)
+				require.NoError(t, err)
+				testWaitForTask(t, i, gotTask)
+			}
 
 			err := i.GetDocuments(tt.args.request, tt.args.resp)
 			require.NoError(t, err)
 			if tt.args.request != nil && tt.args.request.Limit != 0 {
 				require.Equal(t, tt.args.request.Limit, int64(len(tt.args.resp.Results)))
+			} else if tt.result != 0 {
+				require.Equal(t, tt.result, int64(len(tt.args.resp.Results)))
 			} else {
-				require.Equal(t, 6, len(tt.args.resp.Results))
+				require.Equal(t, 20, len(tt.args.resp.Results))
```

Comment:
Why we just don't have a `require.Equal(t, tt.result, int64(len(tt.args.resp.Results)))` directly?

Reply:
Because if the parameters or limit are not changed, there should be 20 results by default, so I haven't set any "result" in these test cases, and I'm verifying that there are indeed the expected default number of results.
I can modify it but I had to set the `results` is all this tests

Reply:
I see the point now. To be honest, I prefer seeing the duplicated values in each test case than understanding business logic in the tests. Tests are meant to be clear, some resources to back that POV:

- https://stackoverflow.com/a/129722/2649707
- https://stackoverflow.com/a/130038/2649707
- https://www.codewithjason.com/why-duplication-is-more-acceptable-in-tests/

Since you already have a setup where you write use cases by defining structs and then iterating over them, you have already reduced a lot of boilerplates. As you probably noted, the go tests differ from all the other SDKs.
I think you could reconsider the decision to keep those conditionals in the test and "duplicate" the setup in another issue.